src/libs/actions/Report.js

```diff
@@ -298,19 +298,26 @@ function fetchIOUReportID(debtorEmail) {
 }
 
 /**
- * Fetches chat reports when provided a list of
- * chat report IDs
- *
+ * Fetches chat reports when provided a list of chat report IDs.
+ * If the shouldRedirectIfInacessible flag is set, we redirect to the Concierge chat
+ * when fetching a single chat that is inacessible.
  * @param {Array} chatList
+ * @param {Boolean} shouldRedirectIfInacessible
  * @returns {Promise<Number[]>} only used internally when fetchAllReports() is called
  */
-function fetchChatReportsByIDs(chatList) {
+function fetchChatReportsByIDs(chatList, shouldRedirectIfInacessible = false) {
     let fetchedReports;
     const simplifiedReports = {};
     return API.GetReportSummaryList({reportIDList: chatList.join(',')})
-        .then(({reportSummaryList}) => {
+        .then(({reportSummaryList, jsonCode}) => {
             Log.info('[Report] successfully fetched report data', true);
             fetchedReports = reportSummaryList;
+
+            // If we receive a 404 response while fetching a single report, treat that report as inacessible.
+            if (jsonCode === 404 && chatList.length === 1) {
```

Comment:
1. not sure checking the `chatList` is necessary on second thought
2. we should be checking `shouldRedirectIfInacessible` when we throw this error

Reply:
Makes sense. If we're controlling the redirect via the `shouldRedirectIfInacessible` param, checking the `chatList` is a weird restriction.